File: src/Api/Models/SocialPlatformKeys.cs
namespace PostGenerator.Api.Models;

/// <summary>Known credential key names per platform for UI and masking.</summary>
public static class SocialPlatformKeys
{
    public static readonly IReadOnlyList<string> AllPlatforms = new[] { "LinkedIn", "Bluesky", "Instagram", "Facebook", "TikTok", "Skool" };

    public static readonly IReadOnlyDictionary<string, string[]> KeysByPlatform = new Dictionary<string, string[]>(StringComparer.OrdinalIgnoreCase)
    {
        ["LinkedIn"] = new[] { "AccessToken", "PersonUrn" },
        ["Bluesky"] = new[] { "Handle", "AppPassword", "PdsUrl" },
        ["Instagram"] = new[] { "UserId", "AccessToken" },
        ["Facebook"] = new[] { "PageId", "PageAccessToken" },
        ["TikTok"] = new[] { "AccessToken" },
        ["Skool"] = new[] { "ApiKey", "SessionId", "GroupId" },
    };
}

File: src/lib/social-platform-keys.ts
export const ALL_PLATFORMS = [
  "LinkedIn",
  "Bluesky",
  "Instagram",
  "Facebook",
  "TikTok",
  "Skool",
] as const;

export const KEYS_BY_PLATFORM: Record<string, string[]> = {
  LinkedIn: ["AccessToken", "PersonUrn"],
  Bluesky: ["Handle", "AppPassword", "PdsUrl"],
  Instagram: ["UserId", "AccessToken"],
  Facebook: ["PageId", "PageAccessToken"],
  TikTok: ["AccessToken"],
  Skool: ["ApiKey", "SessionId", "GroupId"],
};
